### PR TITLE
Various fixes

### LINKED ISSUE #746: leftwm becomes unresponsive after closing spotify-client.
**Describe the bug**
After closing a single instance of the `spotify-client` leftwm becomes unresponsive to any commands.

**To Reproduce**
1. Go to an empty workspace
2. Open `spotify-client` using dmenu / rofi / shortcut
3. Close the window using a shortcut
4. Only the mouse moves but leftwm becomes unresponsive to any keybinds

**Expected behavior**
Continous functionality(?)

**Screenshots**
no way to take them atm. The entire wm becomes unresponsive :c

**Environment:**
 - OS/Distro: Tested and this happens in Debian 11, Void and Ubuntu (20.04)
 - Output of `leftwm-check`
```
:: LeftWM version: 0.3.0
:: LeftWM git hash: 557d09c
:: Loading configuration . . .
    -> Configuration loaded OK 
:: Checking keybinds . . .
    -> All keybinds OK
:: Checking environment . . .
    -> Environment OK 
:: Checking theme . . .
    -> Theme OK 
```

**Additional context**
Idk if a floating window wants to be opened and leaves the wm stuck but doesn't seem to be the case as switching windows doesn't work. Also it is not possible to kill the session so this hangs the wm in some unknown way

### PR DESCRIPTION
**Pull request by AethanFoot**
Stop cursor snapping to windows which want urgency. Also jetbrains is a pain, it sets it so their windows can never have forced focus, but requires that we ask for focus and it sorts it. The border colour was easy enough to fix, however we disable asking windows to take focus for sloppy due to slow applications (#563). I think I need to think longer on how we may want to get around this or find a better solution to #563.

Many thanks.

Edit fixes #746 or not :anger: or yes :partying_face:

### CODE REVIEW

**Comment by AethanFoot** [comment]
Currently we are in between fixing #563 and jetbrains apps, I am just saying if the window doesnt let us focus it then ask to even when in sloppy. For most apps this is fine, we just get #563 with jetbrain apps (but u dont have to click the app for the focus to switch, let me know if the other way is preferred).

To solve this I think it will need to be a bigger thing, where we ignore enter notifies when we are moving windows with xlib.

**Comment by AethanFoot** [comment]
Really struggling with #746, it seems that xlib is locking on the keyboard when spotify closes until we have an update (e.g. focus change, tag change etc.)

**Comment by carrascomj** [comment]
Awesome, this solves #746 for me. Thanks!

**Comment by hertg** [comment]
fixes #735

**Comment by Zwo1in** [comment]
For me fixed #760

**Review by hertg — APPROVED** [review]

File: leftwm-core/src/display_servers/xlib_display_server/event_translate.rs [hunk 1]
```diff
@@ -70,12 +72,26 @@ fn from_destroy_notify(x_event: XEvent) -> Option<DisplayEvent> {
...
 
+fn from_focus_in(x_event: XEvent) -> Option<DisplayEvent> {
+    let xw = x_event.0;
+    let event = xlib::XFocusChangeEvent::from(x_event.1);
+    // Check that if a window is taking focus, that it should be.
+    if xw.focused_window != event.window {
+        let mut never_focus = false;
+        if let Some(hint) = xw.get_wmhints(xw.focused_window) {
+            never_focus = hint.flags & xlib::InputHint != 0 && hint.input == 0;
+        }
```
[the remark this answers is not in the document]
Comment (AethanFoot) [reply leftwm-core/src/display_servers/xlib_display_server/event_translate.rs]: No reason inparticular just hadnt thought of it

File: leftwm-core/src/display_servers/xlib_display_server/event_translate_client_message.rs [hunk 2]
```diff
@@ -52,7 +52,8 @@ pub fn from_event(xw: &XWrap, event: xlib::XClientMessageEvent) -> Option<Displa
         }
     }
     if event.message_type == xw.atoms.NetActiveWindow {
-        return Some(DisplayEvent::HandleWindowFocus(event.window.into()));
+        xw.set_window_urgency(event.window, true);
```
[the remark this answers is not in the document]
Comment (AethanFoot) [reply leftwm-core/src/display_servers/xlib_display_server/event_translate_client_message.rs:55]: I think they are just an overlap between two different specifications

File: leftwm-core/src/display_servers/xlib_display_server/event_translate.rs [hunk 3]
```diff
@@ -70,12 +72,26 @@ fn from_destroy_notify(x_event: XEvent) -> Option<DisplayEvent> {
...
 }
 
+fn from_focus_in(x_event: XEvent) -> Option<DisplayEvent> {
+    let xw = x_event.0;
+    let event = xlib::XFocusChangeEvent::from(x_event.1);
+    // Check that if a window is taking focus, that it should be.
+    if xw.focused_window != event.window {
+        let never_focus = match xw.get_wmhints(xw.focused_window) {
+            Some(hint) => hint.flags & xlib::InputHint != 0 && hint.input == 0,
+            None => false
```
Comment (hertg) [inline leftwm-core/src/display_servers/xlib_display_server/event_translate.rs:88]: Sorry, my suggestion tripped up fmt :smile: 

```suggestion
            None => false,
```

**Review by hertg — APPROVED** [review]

**Pull request merged**